Frends.LDAP.AddUserToGroups/Frends.LDAP.AddUserToGroups.Tests/UnitTests.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Frends.LDAP.AddUserToGroups.Definitions;
+using Novell.Directory.Ldap;
+namespace Frends.LDAP.AddUserToGroups.Tests;
+
+[TestClass]
+public class UnitTests
+{
+    // Azure AD.
+    /*
+        private readonly int _port = 389;
+        private readonly string? _path = "CN=Users,DC=FRENDSTest01,DC=net";
+        private readonly string? _dn = "CN=Tes Tuser,CN=Users,DC=FRENDSTest01,DC=net";
+        private readonly string? _groupDn = "CN=Users,CN=Builtin,DC=FRENDSTest01,DC=net";
+        private readonly string? _host = Environment.GetEnvironmentVariable("HiQ_AzureADTest_Address");
+        private readonly string? _user = Environment.GetEnvironmentVariable("HiQ_AzureADTest_User");
+        private readonly string? _pw = Environment.GetEnvironmentVariable("HiQ_AzureADTest_Password");
+    */
+
+    /*
+        LDAP server to docker.
+        docker run -d -it --rm -p 10389:10389 dwimberger/ldap-ad-it
+    */
+    private readonly string? _host = "127.0.0.1";
+    private readonly int _port = 10389;
+    private readonly string? _user = "uid=admin,ou=system";
+    private readonly string? _pw = "secret";
+    private readonly string _path = "ou=users,dc=wimpi,dc=net";
+    private readonly string? _dn = "CN=Tes Tuser,ou=users,dc=wimpi,dc=net";
+    private readonly string? _groupDn = "cn=admin,ou=roles,dc=wimpi,dc=net";
+
+    Input? input;
+    Connection? connection;
+
+    [TestInitialize]
+    public void StartUp()
+    {
+        try
+        {
+            CreateTestUsers();
+        }
+        catch (Exception)
+        {
+        }
+    }
+
+    [TestCleanup]
+    public void CleanUp()
+    {
+        try
+        {
+            DeleteUser();
+        }
+        catch (Exception)
+        {
+        }
+    }
+
+    [TestMethod]
+    public void Update_HandleLDAPError_Test()
+    {
+        input = new()
+        {
+            UserDistinguishedName = "CN=Common Name,CN=Users,DC=Example,DC=Com",
+            GroupDistinguishedName = "CN=Admins,DC=Example,DC=Com"
+        };
+        connection = new()
+        {
+            Host = _host,
+            User = _user,
+            Password = _pw,
+            SecureSocketLayer = false,
+            Port = _port,
+            TLS = false,
+        };
+
+        var result = LDAP.AddUserToGroups(input, connection);
+        Assert.IsTrue(result.Success.Equals(false) && result.Error.Contains("No Such Object"));
+    }
+
+    [TestMethod]
+    public void AddUserToGroups_Test()
```

Comment:
This test fails if run twice.

Reply:
It seems like dummy server only flags these deleted users because they can't be found with search but they already have all the old memberships when added again in another run. Works fine in Azure.